src/cluster_undersample/interface_mlj.jl

```diff
@@ -0,0 +1,146 @@
+
+### ClusterUndersampler with MLJ Interface
+# interface struct
+mutable struct ClusterUndersampler{T} <: Static
+    mode::AbstractString
+    ratios::T
+    maxiter::Integer
+    rng::Integer
+    try_perserve_type::Bool
+end;
+
+"""
+Initiate a cluster undersampling model with the given hyper-parameters.
+"""
+function ClusterUndersampler(;
+    mode::AbstractString = "nearest",
+    ratios::Union{Nothing, AbstractFloat, Dict{T, <:AbstractFloat}} = 1.0,
+    maxiter::Integer = 100,
+    rng::Integer = 42,
+    try_perserve_type::Bool = true,
+) where {T}
+    model = ClusterUndersampler(mode, ratios, maxiter, rng, try_perserve_type)
+    return model
+end
+
+"""
+Undersample data X, y 
+"""
+function MMI.transform(r::ClusterUndersampler, _, X, y)
+    return cluster_undersample(
+        X,
+        y;
+        mode = r.mode,
+        ratios = r.ratios,
+        maxiter = r.maxiter,
+        rng = r.rng,
+        try_perserve_type = r.try_perserve_type,
+    )
+end
+
+MMI.metadata_pkg(
+    ClusterUndersampler,
+    name = "Imbalance",
+    package_uuid = "c709b415-507b-45b7-9a3d-1767c89fde68",
+    package_url = "https://github.com/JuliaAI/Imbalance.jl",
+    is_pure_julia = true,
+)
+
+MMI.metadata_model(
+    ClusterUndersampler,
+    input_scitype = Union{Table(Continuous), AbstractMatrix{Continuous}},
+    output_scitype = Union{Table(Continuous), AbstractMatrix{Continuous}},
+    target_scitype = AbstractVector,
+    load_path = "Imbalance." * string(ClusterUndersampler),
+)
+function MMI.transform_scitype(s::ClusterUndersampler)
+    return Tuple{
+        Union{Table(Continuous), AbstractMatrix{Continuous}},
+        AbstractVector{<:Finite},
+    }
+end
+
+"""
+$(MMI.doc_header(ClusterUndersampler))
+
+`ClusterUndersampler` implements clustering undersampling as presented in Wei-Chao, L., Chih-Fong, T., Ya-Han, H., & Jing-Shang, J. (2017). 
+  Clustering-based undersampling in class-imbalanced data. Information Sciences, 409–410, 17–26. with K-means as
+  the clustering algorithm.
+
+
+# Training data
+
+In MLJ or MLJBase, wrap the model in a machine by
+    mach = machine(model)
+
+There is no need to provide any data here because the model is a static transformer.
+
+Likewise, there is no need to `fit!(mach)`. 
+
+For default values of the hyper-parameters, model can be constructed by
+    model = ClusterUndersampler()
+    
+
+# Hyperparameters
+
+- `mode::AbstractString="nearest`: If `center` then the undersampled data will consist of the centriods of 
+    each cluster found. Meanwhile, if `nearest` then it will consist of the nearest neighbor of each centroid.
```

Comment:
```suggestion
- `mode::AbstractString="nearest`: If `"center"` then the undersampled data will consist of the centriods of 
    each cluster found; if `"nearest"` then it will consist of the nearest neighbor of each centroid.
```